api/src/test/java/org/openmrs/api/ConceptServiceTest.java

```diff
@@ -3954,4 +3961,51 @@ public void getConceptByReference_shouldReturnNullWhenEitherConceptRefIsInvalidO
 		assertNull(conceptService.getConceptByReference("id, name or map which does not match to any concept"));
 		assertNull(conceptService.getConceptByReference("1000")); //invalid uuid but exists in standardTestDataset
 	}
+
+	/**
+	 * @see ConceptService#getConceptReferenceRangesByConceptId(Integer) 
+	 */
+	@Test
+	public void getConceptReferenceRangeByConceptId_shouldReturnConceptReferencesRangeById() {
+		executeDataSet(CONCEPT_WITH_CONCEPT_REFERENCE_RANGES_XML);
+
+		final Integer EXPECTED_CONCEPT_REFERENCE_RANGE_ID = 3;
+		List<ConceptReferenceRange> conceptReferenceRange = conceptService.getConceptReferenceRangesByConceptId(5089);
+		
+		assertFalse(conceptReferenceRange.isEmpty());
+		
+		assertEquals(EXPECTED_CONCEPT_REFERENCE_RANGE_ID, conceptReferenceRange.get(0).getId());
+	}
+
+	/**
+	 * @see ConceptService#saveConcept(Concept)
+	 */
+	@Test
+	public void saveConcept_shouldSaveANewConceptReferenceRange() {
+		Context.setLocale(Locale.US);
+		ConceptNumeric conceptNumeric = new ConceptNumeric();
+		conceptNumeric.setDatatype(new ConceptDatatype(1));
+		conceptNumeric.setConceptClass(new ConceptClass(1));
+
+		ConceptReferenceRange conceptReferenceRange = new ConceptReferenceRange();
+		conceptReferenceRange.setCriteria("$patient.getAge() >= 1 && $patient.getAge() <= 70");
+		conceptReferenceRange.setConceptNumeric(conceptNumeric);
+		conceptReferenceRange.setHiAbsolute(conceptNumeric.getHiAbsolute());
```

Comment:
Is this equivalent to `conceptReferenceRange.setHiAbsolute(null)`?

Reply:
I have fixed it - it's not null now